### ISSUE TITLE
Reslicing a 3D data at any arbitrary angle

### ISSUE THREAD

**Issue by vikashg** [Project-MONAI/MONAI#3815]
**Is your feature request related to a problem? Please describe.**
Is there a method to reslice a 3D volume along any random plane. This will be really useful for data-augmentation tasks.
**Describe the solution you'd like**
A transform that takes a 3D volume and direction cosines for a plane along which the data needs to be resliced. It should also take the slice thickness for the reslicing operation

**Describe alternatives you've considered**
I have considered using the VTK view port functionality to reslice. But not successful yet.

**Additional context**
@AndreasHeumann worked on Clara-Viz which is now working with monai-deploy. I wonder if it uses a VTK backend for the rendering. It will be great, if he can share some insight on this.

**Comment by Nic-Ma** [comment]
Hi @ericspod @AndreasHeumann ,

Do you have any idea about this question?

Thanks in advance.

**Comment by AndreasHeumann** [comment]
No, Clara Viz is not using a VTK backend for the rendering, it's using our own CUDA based renderer. The renderer includes code to re-orient a 3D volume (swap and flip axes) but not to on a random plane.

**Comment by wyli** [comment]
you can also use `monai.transform.Affine(affine=..., spatial_size=(H, W, 1))`, in this case the input `affine` should be generated beforehand, it can be a transform that combines random aug factors,  direction cosines, and thickness.

**Comment by vikashg** [comment]
Hi @wyli and @Nic-Ma, I tried `RandAffine` transform and it works great if we want to extract the individual slices as you mentioned. But if I have to rotate the whole 3D volume, I feel that can be done by changing the spatial size to something like (110, 120, 120). I tried doing it and the transform works but the image is displaced as shown [here](https://drive.google.com/file/d/1RYiCjMUF4G8V8SHRscs_fHtaqcDWL7rR/view?usp=sharing). I think if I can specify the *center of rotation* in the affine transform it will really help to keep the image in center. How can I specify the center of rotation ? Any thoughts ? Thanks

**Comment by wyli** [comment]
That's great! for `rotate_range` `translate_range` in `RandAffine`, they are random factors relative to the image center. the goal of this transform is mainly to combine different spatial transforms and only  resampling. if you are only interested in rotation, there's a `RandRotate` https://github.com/Project-MONAI/MONAI/blob/c3db337644d9525610d9fab3cfd30f963438dc6a/monai/transforms/spatial/array.py#L945
(I'm converting this to a discussion as it's not about the original topic now.)
